chain-patrol-ts/src/fetcher.ts

```diff
@@ -0,0 +1,92 @@
+import { LRUCache } from "lru-cache";
+import { createAssetListApiOptions, createAssetDetailsApiOptions } from "./utils";
+import { ApiOptions, AssetList, Asset, AssetDetails } from "./types";
+import { ASSET_TYPES } from "./constants";
+
+export default class AssetFetcher {
+  assetListUrl: string;
+  assetDetailsUrl: string;
+  assetStatus: string;
+  private apiKey: string;
+  private readonly MAX_TRIES: number;
+  private assetListCache: LRUCache<string, Asset[]>;
+  private assetDetailsCache: LRUCache<string, AssetDetails>;
+
+  constructor(apiKey: string, assetListUrl: string, assetDetailsUrl: string, assetStatus: string) {
+    this.MAX_TRIES = 3;
```

Comment:
since this isn't this class' property, I think this could be written as a constant variable in the top of the file like 
```
const MAX_FETCH_RETRIES = 3;
```